edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesRuntimeInfoProvider.cs

```diff
@@ -96,7 +96,19 @@ public async Task<IEnumerable<ModuleRuntimeInfo>> GetModules(CancellationToken c
             }
         }
 
-        public Task<Stream> GetModuleLogs(string module, bool follow, Option<int> tail, Option<int> since, CancellationToken cancellationToken) => Task.FromResult(Stream.Null);
+        public async Task<Stream> GetModuleLogs(string module, bool follow, Option<int> tail, Option<int> since, CancellationToken cancellationToken)
+        {
+            int tailLines = tail.OrDefault();
+            int sinceSec = since.OrDefault();
+
+            return await this.client.ReadNamespacedPodLogAsync(
+                module,
+                Constants.K8sNamespace,
+                follow: follow,
+                tailLines: tailLines == default(int) ? null : (int?)tailLines,
```

Comment:
@darobs  @dmolokanov  if either of you can think of a better way to convert Option<int> -> int? ... I am all ears. GetOrElse wont work becasue Option expects a func<int>.

Reply:
I am using default(int) as a sentinal value, but that makes me sad :(